### PR TITLE
Feat/56 configuration for regex inputs

### PR DESCRIPTION
**Pull request by BeierKevin**
# PR description
*Describe your changes in detail here*

The proper connection between backend regex (value|constraint) and the UI has been made. Additionally, in the run view when the start button is pressed a proper output what has been captured is noted down.

# Definition Of Done (DoD)
*This PR can be squashed / merged if*
- [x] a developer is assigned
- [x] the PR is NOT estimated
- [x] the PR is labeled
- [x] the PR is assigned to the current sprint
- [x] a meaningful title has been set according to https://www.conventionalcommits.org/
- [x] the PR is described in detail
- [x] the PR links to an issue
- [x] the PR has been reviewed

*Add additional conditions here if necessary for this PR*

fixes: #56

### CODE REVIEW

File: src/components/LogOutput.vue [hunk 1]
```diff
@@ -1,25 +1,85 @@
...
+// watch for changes in isRunning and start/stop the timer
+watch(isRunning, (newValue) => {
+    if (newValue) {
+        timerId = setTimeout(function tick() {
+            let newValue: MatchedElement = vigad.value
+                .getCaptureArea(0)
+                .getRegexGroups()[0]
+                .getValueRegex()
+                .getLastBestMatch();
+            console.log(Math.floor(Date.now() / 1000));
```
Comment (Claiyc) [inline src/components/LogOutput.vue:56]:
```suggestion
```

File: src/proc/TesseractHandler.ts [hunk 2]
```diff
@@ -92,33 +97,55 @@ export class TesseractHandler {
...
+                                    .addJob('recognize', img, { rectangle })
+                                    .then((x) =>
+                                        results.push({
+                                            ca_id: rectangle.id,
+                                            data: x.data.text,
+                                        })
+                                    )
+                            )
+                        );
+                        console.log(results);
```
Comment (Claiyc) [inline src/proc/TesseractHandler.ts:143]:
```suggestion
```

File: src/proc/regex/Regex.ts [hunk 3]
```diff
@@ -139,12 +154,16 @@ export abstract class Regex {
...
      * @return {index: number, element: string}[]
      */
-    private getAllSubstrings(str:string): {index: number, element: string}[] {
-        var i, j, result = [];
+    private getAllSubstrings(
+        str: string
+    ): { index: number; element: string }[] {
+        var i,
+            j,
+            result = [];
```
Comment (Claiyc) [inline src/proc/regex/Regex.ts:162]: this is ugly as hell

**Review by Claiyc — CHANGES_REQUESTED** [review]
Timestamp does not work (timestamp for older log entrys gets replaced for every new one that's being added):
grafik
(notice how the time is always the same)

Also we shouldn't display anything if no value is found (check for rating = -1)

**Comment by BeierKevin** [comment]
Why should be display nothing if nothing is matched, via the rating -1 the user knows that something is wrong with the regex

File: src/components/capture-area/CaptureAreaSearchValue.vue [hunk 4]
```diff
@@ -39,14 +18,54 @@
...
+    );
+
+    console.log(
+        'After',
+        vigad.value
+            .getCaptureArea(props.captureAreaId)
+            .getRegexGroups()[0]
+            .getConstraintRegex()[1]
+    );
+});
```
Comment (Claiyc) [inline src/components/capture-area/CaptureAreaSearchValue.vue:68]:
```suggestion
```

File: src/proc/regex/Regex.ts [hunk 5]
```diff
@@ -130,7 +145,7 @@ export abstract class Regex {
                 // i--; May be a good performance optimization, doesn't work however if max is greater than the number of possible matches
             }
         }
-        console.log("generated matches for regex: " + this.id);
+        console.log('generated matches for regex: ' + this.id);
```
Comment (Claiyc) [inline src/proc/regex/Regex.ts:148]:
```suggestion
```

**Review by Claiyc — CHANGES_REQUESTED** [review]
Timestamp still wrongly updated.

Regex matching does not seem to work as expected. See #66

File: src/components/capture-area/CaptureAreaSearchValue.vue (hunk 4 above)
Comment (BeierKevin) [reply src/components/capture-area/CaptureAreaSearchValue.vue:68]: Haven't fixed the timestamps yet

**Review by Claiyc — APPROVED** [review]

**Pull request merged**